### PR TITLE
[1.11.x] [RHPAM-4129] - Kogito Operator scorecard errors

### PR DESCRIPTION
**Pull request by spolti**
This commit includes
- scorecard tests fix
- go lang upgrade to 1.15.x
- operator-sdk upgrade to 1.3.2
- replacement of process to rules examples [BAPL-2260]

See: https://issues.redhat.com/browse/RHPAM-4129

Signed-off-by: spolti <fspolti@redhat.com>

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-operator/blob/main/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO|RHPAM-XYZ] Subject`
- [x] Pull Request contains a link to the JIRA issue
- [x] Pull Request contains a description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster
- [x] You've added a [RELEASE_NOTES.md](RELEASE_NOTES.md) entry regarding this change

### CODE REVIEW

**Review by vaibhavjainwiz — APPROVED** [review]

**Comment by radtriste** [comment]
changing golang version and operator-sdk would mean also some changes in our pipelines as well as in prod pipeline ... for the rebuild of 7.12.0 images, I would avoid that and keep that for 7.12.1...

File: bundle/manifests/rhpam-kogito-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -413,6 +627,4 @@ spec:
   maturity: beta
   provider:
     name: Red Hat
-  replaces: rhpam-kogito-operator.v7.11.1-1
-  version: 7.12.0-1
-
+  version: 7.12.0
```
Comment (sutaakar) [inline bundle/manifests/rhpam-kogito-operator.clusterserviceversion.yaml]: I think the old changes should be kept.

**Review by sutaakar — APPROVED** [review]

**Comment by spolti** [comment]
> changing golang version and operator-sdk would mean also some changes in our pipelines as well as in prod pipeline ... for the rebuild of 7.12.0 images, I would avoid that and keep that for 7.12.1...

I have updated the PR, please take a look again.

File: bundle/manifests/rhpam-kogito-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (spolti) [reply bundle/manifests/rhpam-kogito-operator.clusterserviceversion.yaml]: agreed. thanks for pointing this out.

**Review by sutaakar — APPROVED** [review]

**Review by ricardozanini — APPROVED** [review]

**Comment by spolti** [comment]
closed in favor of https://github.com/kiegroup/kogito-operator/pull/1150

**Pull request closed without merging**